<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Tina Lasisi" />


<title>Biological Significance</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/anchor-sections-1.0/anchor-sections.css" rel="stylesheet" />
<script src="site_libs/anchor-sections-1.0/anchor-sections.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Hair Phenotyping Methods Project</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
<li>
  <a href="validation.html">Validation</a>
</li>
<li>
  <a href="significance.html">Significance</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/tinalasisi/2020_HairPheno_manuscript">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Biological Significance</h1>
<h4 class="author">Tina Lasisi</h4>
<h4 class="date">2021-05-05 00:24:25</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-05-05
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>2020_HairPheno_manuscript/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed12345code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(12345)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed12345code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(12345)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomtinalasisi2020HairPhenomanuscripttreeab8587ba8e348c1ee96701e87a7778d0d3616c5etargetblankab8587ba"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/tinalasisi/2020_HairPheno_manuscript/tree/ab8587ba8e348c1ee96701e87a7778d0d3616c5e" target="_blank">ab8587b</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomtinalasisi2020HairPhenomanuscripttreeab8587ba8e348c1ee96701e87a7778d0d3616c5etargetblankab8587ba" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/tinalasisi/2020_HairPheno_manuscript/tree/ab8587ba8e348c1ee96701e87a7778d0d3616c5e" target="_blank">ab8587b</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rproj.user/

Unstaged changes:
    Modified:   analysis/significance.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/significance.Rmd</code>) and HTML (<code>docs/significance.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tinalasisi/2020_HairPheno_manuscript/a0b6c5e6b43f32cb87b9fd61e86b35f4692d6e37/docs/significance.html" target="_blank">a0b6c5e</a>
</td>
<td>
tinalasisi
</td>
<td>
2020-12-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tinalasisi/2020_HairPheno_manuscript/blob/b7cddab9c53f3d0930fc74982ded291bb5e7a6ee/analysis/significance.Rmd" target="_blank">b7cddab</a>
</td>
<td>
tinalasisi
</td>
<td>
2020-12-20
</td>
<td>
wflow_publish(“analysis/significance.Rmd”, update = TRUE)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tinalasisi/2020_HairPheno_manuscript/650ebff1c2de0e9fd70ec80426deda6d46ef59bc/docs/significance.html" target="_blank">650ebff</a>
</td>
<td>
tinalasisi
</td>
<td>
2020-12-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tinalasisi/2020_HairPheno_manuscript/blob/5f6c13808bbf0c07bb670ef913bfcd4b794a9765/analysis/significance.Rmd" target="_blank">5f6c138</a>
</td>
<td>
tinalasisi
</td>
<td>
2020-12-20
</td>
<td>
wflow_publish("analysis/*", update = TRUE)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tinalasisi/2020_HairPheno_manuscript/68f0ec2cce35a807b313cdbd108d9203de4775be/docs/significance.html" target="_blank">68f0ec2</a>
</td>
<td>
tinalasisi
</td>
<td>
2020-11-24
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tinalasisi/2020_HairPheno_manuscript/blob/7ddd3f7f1e18eb327108f32dd2765c4f623717f8/analysis/significance.Rmd" target="_blank">7ddd3f7</a>
</td>
<td>
tinalasisi
</td>
<td>
2020-11-24
</td>
<td>
Initial commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tinalasisi/2020_HairPheno_manuscript/7ddd3f7f1e18eb327108f32dd2765c4f623717f8/docs/significance.html" target="_blank">7ddd3f7</a>
</td>
<td>
tinalasisi
</td>
<td>
2020-11-24
</td>
<td>
Initial commit
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>To explore the significance of quantifying hair fiber morphology, we explore the relationship between various quantitative hair traits, categorical data and genotype data on the same sample.</p>
<p>Our data consists of 193 individuals for whom we have quantitative hair phenotype data. In our first data quality control step, we filter to keep individuals who have more than 4 hair fragments in their curvature image and over 10% African ancestry. We calculate mean and median values for the cross-sectional data we have collected for individuals (~ 6 sectioned hair fibers). In our analyses, we use median values as they are less affected by intra-individual outliers.</p>
<div id="self-reported-hair-texture-vs.-quantitative-hair-curvature" class="section level1">
<h1><span class="header-section-number">1</span> Self-reported hair texture vs. quantitative hair curvature</h1>
<p>We compare the self-reported hair texture with mean and median curvature for our sample.</p>
<div class="figure" style="text-align: center">
<img src="figure/significance.Rmd/plt_admixed_SelfRepHair-1.png" alt="Self-reported hair texture vs. quantitative hair curvature" width="100%" />
<p class="caption">
Self-reported hair texture vs. quantitative hair curvature
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plt_admixed_SelfRepHair-1">
Past versions of plt_admixed_SelfRepHair-1.png
</button>
</p>
<div id="fig-plt_admixed_SelfRepHair-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tinalasisi/2020_HairPheno_manuscript/blob/7ddd3f7f1e18eb327108f32dd2765c4f623717f8/docs/figure/significance.Rmd/plt_admixed_SelfRepHair-1.png" target="_blank">7ddd3f7</a>
</td>
<td>
tinalasisi
</td>
<td>
2020-11-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>The single individual with a high mean curvature in the straight group is the result of an artefact in the image.</p>
<div class="figure" style="text-align: center">
<img src="assets/artefact.jpg" alt="Image of hair sample with artefact biasing the measurement" width="100%" />
<p class="caption">
Image of hair sample with artefact biasing the measurement
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-artefact">
Past versions of artefact.jpg
</button>
</p>
<div id="fig-artefact" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tinalasisi/2020_HairPheno_manuscript/blob/7ddd3f7f1e18eb327108f32dd2765c4f623717f8/docs/assets/artefact.jpg" target="_blank">7ddd3f7</a>
</td>
<td>
tinalasisi
</td>
<td>
2020-11-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>The red arrow points to a stray fiber that likely contaminated the sample and was missed during imaging. Such potential outliers are the reason we chose to use the median curvature for a sample in our analyses.</p>
</div>
<div id="objective-hair-texture-classification-vs.-quantitative-curvature" class="section level1">
<h1><span class="header-section-number">2</span> Objective hair texture classification vs. quantitative curvature</h1>
<p>To explore how much data is lost when binning continuous variation, we compared mean and median curvature to classified hair texture. This classification is based on Loussouarn et al.’s 2007 paper <a href="https://doi.org/10.1111/j.1365-4632.2007.03453.x">"Worldwide diversity of hair curliness: a new method of assessment</a>.</p>
<p>While the authors propose a number of parameters to distinguish curlier hair types (based on number of twists and waves among other factors), their primary classification is based on curvature. We demonstrate that, regardless of additional parameters, a considerable range of curvature is obscured when collapsing hair variation according to their curvature thresholds.</p>
<div class="figure" style="text-align: center">
<img src="figure/significance.Rmd/plt_admixed_HairTypeBin-1.png" alt="Objective hair classification vs. quantitative curvature" width="100%" />
<p class="caption">
Objective hair classification vs. quantitative curvature
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plt_admixed_HairTypeBin-1">
Past versions of plt_admixed_HairTypeBin-1.png
</button>
</p>
<div id="fig-plt_admixed_HairTypeBin-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tinalasisi/2020_HairPheno_manuscript/blob/7ddd3f7f1e18eb327108f32dd2765c4f623717f8/docs/figure/significance.Rmd/plt_admixed_HairTypeBin-1.png" target="_blank">7ddd3f7</a>
</td>
<td>
tinalasisi
</td>
<td>
2020-11-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="ancestry-vs.-hair-morphology" class="section level1">
<h1><span class="header-section-number">3</span> Ancestry vs. hair morphology</h1>
<p>We carried out a number of analyses using the genotype data collected for this diverse sample. In an admixed sample where a continuous trait has divergent distributions in the parental ancestry groups, the resulting admixed population can show a correlation between ancestry and that trait. Finding such a correlation suggests may imply a polygenic trait with high heritability.</p>
<div id="admixture-components" class="section level2">
<h2><span class="header-section-number">3.1</span> Admixture components</h2>
<p>Our sample consists of admixed individuals with primarily African and European ancestry.</p>
<div class="figure" style="text-align: center">
<img src="figure/significance.Rmd/plt_admixture-1.png" alt="Admixture components for sample" width="100%" />
<p class="caption">
Admixture components for sample
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plt_admixture-1">
Past versions of plt_admixture-1.png
</button>
</p>
<div id="fig-plt_admixture-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tinalasisi/2020_HairPheno_manuscript/blob/7ddd3f7f1e18eb327108f32dd2765c4f623717f8/docs/figure/significance.Rmd/plt_admixture-1.png" target="_blank">7ddd3f7</a>
</td>
<td>
tinalasisi
</td>
<td>
2020-11-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>The colors represent ancestries that correspond to the following 1000 Genomes populations: - SAS = South Asian - AMR = American - AFR = African - EUR = European - EAS = East Asian</p>
<p>Each of these are metapopulations based on the grouping of multiple (sub)continental population groups in the <a href="https://www.internationalgenome.org/category/population/">1000 Genomes repository</a>.</p>
</div>
<div id="ancestry-vs.-curvature" class="section level2">
<h2><span class="header-section-number">3.2</span> Ancestry vs. curvature</h2>
<p>Here we plot the correlation between proportion of African ancestry and m-index, median curvature, and eccentricity.</p>
<div class="figure" style="text-align: center">
<img src="figure/significance.Rmd/plt_anc_corr-1.png" alt="Percentage of African ancestry vs. M-index, curvature and eccentricity" width="100%" />
<p class="caption">
Percentage of African ancestry vs. M-index, curvature and eccentricity
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plt_anc_corr-1">
Past versions of plt_anc_corr-1.png
</button>
</p>
<div id="fig-plt_anc_corr-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tinalasisi/2020_HairPheno_manuscript/blob/7ddd3f7f1e18eb327108f32dd2765c4f623717f8/docs/figure/significance.Rmd/plt_anc_corr-1.png" target="_blank">7ddd3f7</a>
</td>
<td>
tinalasisi
</td>
<td>
2020-11-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="curvature-vs.-eccentricity" class="section level2">
<h2><span class="header-section-number">3.3</span> Curvature vs. eccentricity</h2>
<p>The relationship between cross-sectional shape (eccentricity) and curvature has long been debated. Due to the coincidence of cross-sectional shape and curvature in various populations that are often contrasted (i.e. East Asian vs. North European vs. West African), it has been unclear whether these traits have a causal relationship (specifically that higher eccentricity predicts higher curvature). In our admixed sample, we have the opportunity to test this question and fit a model between these traits with and without ancestry.</p>
<div id="uncorrected" class="section level3">
<h3><span class="header-section-number">3.3.1</span> Uncorrected</h3>
<p>First we examine the data without correcting for ancestry.</p>
<pre><code>
Call:
lm(formula = curv_median ~ eccentricity_median, data = df_curv_ecc)

Residuals:
    Min      1Q  Median      3Q     Max 
-0.2843 -0.1245 -0.0264  0.1125  0.4770 

Coefficients:
                    Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)          -0.5682     0.1270  -4.474 1.94e-05 ***
eccentricity_median   1.0284     0.1692   6.076 1.96e-08 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 0.1576 on 106 degrees of freedom
Multiple R-squared:  0.2583,    Adjusted R-squared:  0.2513 
F-statistic: 36.92 on 1 and 106 DF,  p-value: 1.959e-08</code></pre>
<p><img src="figure/significance.Rmd/stats_curv_ecc-1.png" width="100%" style="display: block; margin: auto;" /><img src="figure/significance.Rmd/stats_curv_ecc-2.png" width="100%" style="display: block; margin: auto;" /><img src="figure/significance.Rmd/stats_curv_ecc-3.png" width="100%" style="display: block; margin: auto;" /><img src="figure/significance.Rmd/stats_curv_ecc-4.png" width="100%" style="display: block; margin: auto;" /></p>
<div class="figure" style="text-align: center">
<img src="figure/significance.Rmd/plt_curv_ecc-1.png" alt="Curvature vs. eccentricity (without correction for ancestry)" width="100%" />
<p class="caption">
Curvature vs. eccentricity (without correction for ancestry)
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plt_curv_ecc-1">
Past versions of plt_curv_ecc-1.png
</button>
</p>
<div id="fig-plt_curv_ecc-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tinalasisi/2020_HairPheno_manuscript/blob/7ddd3f7f1e18eb327108f32dd2765c4f623717f8/docs/figure/significance.Rmd/plt_curv_ecc-1.png" target="_blank">7ddd3f7</a>
</td>
<td>
tinalasisi
</td>
<td>
2020-11-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>The residual plots show that there might be a few problems with the relationship: 1. possibly an outlier distorting the relationship, 2. relationship might not be linear 3. errors might be heteroescedastic.</p>
<p>Let’s refit the model after removing the outlier.</p>
<pre><code>
Call:
lm(formula = curv_median ~ eccentricity_median, data = df_curv_ecc)

Residuals:
     Min       1Q   Median       3Q      Max 
-0.29590 -0.12290 -0.01791  0.09892  0.46713 

Coefficients:
                    Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)          -0.6707     0.1387  -4.836 4.56e-06 ***
eccentricity_median   1.1622     0.1842   6.311 6.73e-09 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 0.1561 on 105 degrees of freedom
Multiple R-squared:  0.275, Adjusted R-squared:  0.2681 
F-statistic: 39.83 on 1 and 105 DF,  p-value: 6.731e-09</code></pre>
<p><img src="figure/significance.Rmd/unnamed-chunk-1-1.png" width="100%" style="display: block; margin: auto;" /><img src="figure/significance.Rmd/unnamed-chunk-1-2.png" width="100%" style="display: block; margin: auto;" /><img src="figure/significance.Rmd/unnamed-chunk-1-3.png" width="100%" style="display: block; margin: auto;" /><img src="figure/significance.Rmd/unnamed-chunk-1-4.png" width="100%" style="display: block; margin: auto;" /></p>
<p>Next, let’s see if the data fits a quadratic model better.</p>
<pre><code>Analysis of Variance Table

Model 1: curv_median ~ eccentricity_median
Model 2: curv_median ~ eccentricity_median + eccentricity_median2
  Res.Df    RSS Df Sum of Sq Pr(&gt;Chi)   
1    105 2.5589                         
2    104 2.3655  1   0.19343 0.003543 **
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<p><img src="figure/significance.Rmd/unnamed-chunk-2-1.png" width="100%" style="display: block; margin: auto;" /><img src="figure/significance.Rmd/unnamed-chunk-2-2.png" width="100%" style="display: block; margin: auto;" /><img src="figure/significance.Rmd/unnamed-chunk-2-3.png" width="100%" style="display: block; margin: auto;" /><img src="figure/significance.Rmd/unnamed-chunk-2-4.png" width="100%" style="display: block; margin: auto;" /></p>
<p>Finally, let’s use the fitted values as weights in a weighted regression. Because the variance increases with increasing eccentricity, we will use 1/fitted_values^2 as weights.</p>
<pre><code>
Call:
lm(formula = curv_median ~ eccentricity_median + eccentricity_median2, 
    data = df_curv_ecc, weights = inv.curv)

Weighted Residuals:
    Min      1Q  Median      3Q     Max 
-0.9321 -0.6342 -0.1648  0.3443  3.1744 

Coefficients:
                     Estimate Std. Error t value Pr(&gt;|t|)   
(Intercept)            1.9881     0.7324   2.714  0.00777 **
eccentricity_median   -6.2326     2.1458  -2.905  0.00449 **
eccentricity_median2   5.0686     1.5617   3.245  0.00158 **
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 0.8123 on 104 degrees of freedom
Multiple R-squared:  0.2557,    Adjusted R-squared:  0.2414 
F-statistic: 17.87 on 2 and 104 DF,  p-value: 2.14e-07</code></pre>
<p><img src="figure/significance.Rmd/unnamed-chunk-3-1.png" width="100%" style="display: block; margin: auto;" /><img src="figure/significance.Rmd/unnamed-chunk-3-2.png" width="100%" style="display: block; margin: auto;" /><img src="figure/significance.Rmd/unnamed-chunk-3-3.png" width="100%" style="display: block; margin: auto;" /><img src="figure/significance.Rmd/unnamed-chunk-3-4.png" width="100%" style="display: block; margin: auto;" /></p>
<p>Now the residuals look a lot better so we will go with this model. Let’s generate the scatterplot with the model fit.</p>
<p><img src="figure/significance.Rmd/unnamed-chunk-4-1.png" width="100%" style="display: block; margin: auto;" /></p>
<p>If we consider the relationship between curvature and eccentricity without taking into account ancestry, we find that eccentricity is a significant predictor of curvature.</p>
</div>
<div id="corrected" class="section level3">
<h3><span class="header-section-number">3.3.2</span> Corrected</h3>
<p>We then re-plot the relationship after residualizing curvature and eccentriciy (both) on ancestry.</p>
<div class="figure" style="text-align: center">
<img src="figure/significance.Rmd/plt_curv_ecc_anc-1.png" alt="Curvature vs. eccentricity (with correction for ancestry)" width="100%" />
<p class="caption">
Curvature vs. eccentricity (with correction for ancestry)
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plt_curv_ecc_anc-1">
Past versions of plt_curv_ecc_anc-1.png
</button>
</p>
<div id="fig-plt_curv_ecc_anc-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tinalasisi/2020_HairPheno_manuscript/blob/7ddd3f7f1e18eb327108f32dd2765c4f623717f8/docs/figure/significance.Rmd/plt_curv_ecc_anc-1.png" target="_blank">7ddd3f7</a>
</td>
<td>
tinalasisi
</td>
<td>
2020-11-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>However, when we correct for ancestry, this correlation is no longer significant. This supports the idea that these traits may be independent. Let’s fit a formal model here.</p>
<p>The relationship looks rather linear so let’s start with a simple linear model and inspect the residuals.</p>
<pre><code>
Call:
lm(formula = curv_median ~ eccentricity_median + AFR, data = df_curv_ecc)

Residuals:
     Min       1Q   Median       3Q      Max 
-0.37867 -0.04211 -0.00799  0.03639  0.29942 

Coefficients:
                    Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)         -0.08587    0.09768  -0.879    0.381    
eccentricity_median  0.12775    0.14012   0.912    0.364    
AFR                  0.47886    0.03716  12.885   &lt;2e-16 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 0.09735 on 104 degrees of freedom
Multiple R-squared:  0.7208,    Adjusted R-squared:  0.7154 
F-statistic: 134.2 on 2 and 104 DF,  p-value: &lt; 2.2e-16</code></pre>
<p><img src="figure/significance.Rmd/unnamed-chunk-5-1.png" width="100%" style="display: block; margin: auto;" /><img src="figure/significance.Rmd/unnamed-chunk-5-2.png" width="100%" style="display: block; margin: auto;" /><img src="figure/significance.Rmd/unnamed-chunk-5-3.png" width="100%" style="display: block; margin: auto;" /><img src="figure/significance.Rmd/unnamed-chunk-5-4.png" width="100%" style="display: block; margin: auto;" /></p>
<p>The relationship looks pretty linear but there’s still some heteroescedasticity in the residuals. We will fit a weighted regression model.</p>
<pre><code>
Call:
lm(formula = curv_median ~ eccentricity_median + AFR, data = df_curv_ecc, 
    weights = fitted.cve_afr)

Weighted Residuals:
     Min       1Q   Median       3Q      Max 
-0.90751 -0.40872 -0.06239  0.25190  2.43315 

Coefficients:
                    Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)         -0.02120    0.03712  -0.571    0.569    
eccentricity_median  0.03448    0.05568   0.619    0.537    
AFR                  0.48972    0.05118   9.570 6.17e-16 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 0.5622 on 104 degrees of freedom
Multiple R-squared:  0.4921,    Adjusted R-squared:  0.4824 
F-statistic: 50.39 on 2 and 104 DF,  p-value: 5.003e-16</code></pre>
<p><img src="figure/significance.Rmd/unnamed-chunk-6-1.png" width="100%" style="display: block; margin: auto;" /></p>
<p>The weighted regression corrects the heteroescedasticity well.</p>
<p><img src="figure/significance.Rmd/unnamed-chunk-7-1.png" width="100%" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="curvature-vs.-skin-pigmentation" class="section level2">
<h2><span class="header-section-number">3.4</span> Curvature vs. skin pigmentation</h2>
<p>To demonstrate the potential effect of population stratification on traits, we compare hair curvature with skin pigmentation (m-index). These two traits are not biologically related, yet, in an admixed population, we may see a correlation that is due to population stratification of these polygenic traits.</p>
<div id="uncorrected-1" class="section level3">
<h3><span class="header-section-number">3.4.1</span> Uncorrected</h3>
<p>First we examine the relationship between curvature and skin pigmentation without correcting for ancestry.</p>
<pre><code>
Call:
lm(formula = curv_median ~ m_index, data = df_curv_mindex)

Residuals:
     Min       1Q   Median       3Q      Max 
-0.20715 -0.06804 -0.03471  0.04818  0.51483 

Coefficients:
             Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept) -0.280232   0.041533  -6.747 4.91e-10 ***
m_index      0.012887   0.001031  12.501  &lt; 2e-16 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 0.1261 on 126 degrees of freedom
Multiple R-squared:  0.5536,    Adjusted R-squared:  0.5501 
F-statistic: 156.3 on 1 and 126 DF,  p-value: &lt; 2.2e-16</code></pre>
<p><img src="figure/significance.Rmd/stats_curv_mindex-1.png" width="100%" style="display: block; margin: auto;" /><img src="figure/significance.Rmd/stats_curv_mindex-2.png" width="100%" style="display: block; margin: auto;" /><img src="figure/significance.Rmd/stats_curv_mindex-3.png" width="100%" style="display: block; margin: auto;" /><img src="figure/significance.Rmd/stats_curv_mindex-4.png" width="100%" style="display: block; margin: auto;" /></p>
<div class="figure" style="text-align: center">
<img src="figure/significance.Rmd/plt_curv_mindex-1.png" alt="Curvature vs. M-index (without correction for ancestry)" width="100%" />
<p class="caption">
Curvature vs. M-index (without correction for ancestry)
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plt_curv_mindex-1">
Past versions of plt_curv_mindex-1.png
</button>
</p>
<div id="fig-plt_curv_mindex-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tinalasisi/2020_HairPheno_manuscript/blob/7ddd3f7f1e18eb327108f32dd2765c4f623717f8/docs/figure/significance.Rmd/plt_curv_mindex-1.png" target="_blank">7ddd3f7</a>
</td>
<td>
tinalasisi
</td>
<td>
2020-11-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>As expected, we see a significant correlation between the two traits. The relationship appears to be linear. There still seems to be more variance for observations with higher melanin index/curvature. We will use a weighted regression approach again.</p>
<pre><code>
Call:
lm(formula = curv_median ~ m_index, data = df_curv_mindex, weights = weights_cvm)

Weighted Residuals:
    Min      1Q  Median      3Q     Max 
-0.8842 -0.3855 -0.2268  0.1617  4.4355 

Coefficients:
             Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept) -0.312749   0.047890  -6.531 1.46e-09 ***
m_index      0.013704   0.001582   8.660 1.92e-14 ***
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 0.7457 on 126 degrees of freedom
Multiple R-squared:  0.3731,    Adjusted R-squared:  0.3681 
F-statistic: 74.99 on 1 and 126 DF,  p-value: 1.917e-14</code></pre>
<p><img src="figure/significance.Rmd/unnamed-chunk-8-1.png" width="100%" style="display: block; margin: auto;" /></p>
<p>That fixes the heteroescedasticity issue.</p>
<p><img src="figure/significance.Rmd/unnamed-chunk-9-1.png" width="100%" style="display: block; margin: auto;" /></p>
</div>
<div id="corrected-1" class="section level3">
<h3><span class="header-section-number">3.4.2</span> Corrected</h3>
<p>We then apply a correction for ancestry and re-analyze the data.</p>
<p><img src="figure/significance.Rmd/stats_curv_mindex_anc-1.png" width="100%" style="display: block; margin: auto;" /></p>
<p>Like with curvature and eccentricity, the relationship between curvature and skin pigmentation is no longer significant when ancestry is taken into account.</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre><code>R version 4.0.3 (2020-10-10)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Catalina 10.15.7

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] cowplot_1.1.0   knitr_1.30      forcats_0.5.0   stringr_1.4.0  
 [5] dplyr_1.0.2     purrr_0.3.4     readr_1.4.0     tidyr_1.1.2    
 [9] tibble_3.0.4    ggplot2_3.3.2   tidyverse_1.3.0

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.5        lattice_0.20-41   lubridate_1.7.9.2 assertthat_0.2.1 
 [5] rprojroot_1.3-2   digest_0.6.27     R6_2.5.0          cellranger_1.1.0 
 [9] plyr_1.8.6        backports_1.1.10  reprex_0.3.0      evaluate_0.14    
[13] httr_1.4.2        highr_0.8         pillar_1.4.6      rlang_0.4.8      
[17] readxl_1.3.1      rstudioapi_0.11   whisker_0.4       Matrix_1.2-18    
[21] rmarkdown_2.5     splines_4.0.3     labeling_0.4.2    munsell_0.5.0    
[25] broom_0.7.2       compiler_4.0.3    httpuv_1.5.4      modelr_0.1.8     
[29] xfun_0.19         pkgconfig_2.0.3   mgcv_1.8-33       htmltools_0.5.0  
[33] tidyselect_1.1.0  workflowr_1.6.2   fansi_0.4.1       crayon_1.3.4     
[37] dbplyr_2.0.0      withr_2.3.0       later_1.1.0.1     grid_4.0.3       
[41] nlme_3.1-149      jsonlite_1.7.1    gtable_0.3.0      lifecycle_0.2.0  
[45] DBI_1.1.0         git2r_0.27.1      magrittr_1.5      scales_1.1.1     
[49] cli_2.1.0         stringi_1.5.3     farver_2.0.3      reshape2_1.4.4   
[53] fs_1.5.0          promises_1.1.1    xml2_1.3.2        ellipsis_0.3.1   
[57] generics_0.1.0    vctrs_0.3.4       tools_4.0.3       glue_1.4.2       
[61] hms_0.5.3         yaml_2.2.1        colorspace_1.4-1  rvest_0.3.6      
[65] haven_2.3.1      </code></pre>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
